app/controllers/samvera/persona/users_controller.rb

```diff
@@ -42,6 +42,84 @@ def index
       @presenter = Samvera::Persona::UsersPresenter.new
     end
 
+    # POST /persona/users/paged_index
+    def paged_index
+      @presenter = Samvera::Persona::UsersPresenter.new
+      records_total = @presenter.user_count
+      @presenter = @presenter.users
+      columns = ['username', 'email', 'entry', 'last_sign_in_at', 'invitation_token', 'provider', 'actions']
+
+      # Filtering
+      search_value = params['search']['value']
+      @presenter = if search_value.present?
+                     search_role = @presenter.select { |p| p.groups.any? { |g| g.include? search_value } }
+                     search_date = @presenter.select { |p| last_sign_in(p).to_formatted_s(:long_ordinal).include? search_value }
+                     search_status = @presenter.select { |p| user_status(p).downcase.include? search_value.downcase }
```

Comment:
I'm a little concerned about how this scales, but we could try deploying it and see if it is really an issue or not.